analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -30,6 +32,32 @@ public class NullPointerDereference : NullPointerDereferenceBase
 
         protected override DiagnosticDescriptor Rule => S2259;
 
-        public override bool ShouldExecute() => true;    // ToDo: Implement rule-specific logic
+        public override bool ShouldExecute()
+        {
+            var walker = new SyntaxKindWalker();
+            walker.SafeVisit(Node);
+            return walker.Result;
+        }
+
+        private sealed class SyntaxKindWalker : SafeVisualBasicSyntaxWalker
+        {
+            public bool Result { get; private set; }
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (node.IsAnyKind(
+                    SyntaxKind.AwaitExpression,
+                    SyntaxKind.ForEachStatement,
+                    SyntaxKind.InvocationExpression,    // For array access arr(42)
+                    SyntaxKind.SimpleMemberAccessExpression))
```

Comment:
If you add Throw in C# it needs to be added here too.

Reply:
Same as C#, not yet